README.md
# Overview

Pantomime is replacement for the MIME-manipulation features of [JavaMail](http://www.oracle.com/technetwork/java/javamail/index.html).

Pantomime is:

- robust
- memory efficient
- easier to use.

Download Pantomime from <https://bitbucket.org/barbee/pantomime/downloads>.

Pantomime is licensed under [Mozilla Public License v 2.0](http://mozilla.org/MPL/2.0/)

Pantomime is written by [JH Barbee](http://www.linkedin.com/in/jhbarbee)

# Version

The current version is 0.99. It is in beta.

# Platforms

Pantomime runs in a Java Virtual Machine.

# Installation

1. Download the Pantomime package from <https://bitbucket.org/barbee/pantomime/downloads>.
2. Copy it to your classpath next to your JavaMail jar.

# Issues

Please visit the project home page at  <https://bitbucket.org/barbee/pantomime> to file a bug.


history.txt
? notCheckedIn

RCS file: /home/barbee/CVS/pantomime/.cvsignore,v
Working file: .cvsignore
head: 1.2
branch:
locks: strict
access list:
symbolic names:
	start: 1.1.1.1
	barbee: 1.1.1
keyword substitution: kv
total revisions: 3;	selected revisions: 3
description:
----------------------------
revision 1.2
date: 2013/08/23 21:16:14;  author: barbee;  state: Exp;  lines: +1 -0
Done with 2007.
----------------------------
revision 1.1
date: 2013/08/21 19:25:23;  author: barbee;  state: Exp;
branches:  1.1.1;
Initial revision
----------------------------
revision 1.1.1.1
date: 2013/08/21 19:25:23;  author: barbee;  state: Exp;  lines: +0 -0
New project.
=============================================================================

RCS file: /home/barbee/CVS/pantomime/LICENSE,v
Working file: LICENSE
head: 1.1
branch:
locks: strict
access list:
symbolic names:
keyword substitution: kv
total revisions: 1;	selected revisions: 1
description:
----------------------------
revision 1.1
date: 2013/08/27 16:39:17;  author: barbee;  state: Exp;
Added licensing.
Version is now 0.25.
=============================================================================

RCS file: /home/barbee/CVS/pantomime/README.md,v
Working file: README.md
head: 1.1
branch:
locks: strict
access list:
symbolic names:
keyword substitution: kv
total revisions: 1;	selected revisions: 1
description:
----------------------------
revision 1.1
date: 2013/08/27 16:45:10;  author: barbee;  state: Exp;
Added readme for bitbucket.
=============================================================================

RCS file: /home/barbee/CVS/pantomime/build.gradle,v
Working file: build.gradle
head: 1.16
branch:
locks: strict
access list:
symbolic names:
	start: 1.1.1.1
	barbee: 1.1.1
keyword substitution: kv
total revisions: 17;	selected revisions: 17
description:
----------------------------
revision 1.16
date: 2015/05/08 15:12:09;  author: barbee;  state: Exp;  lines: +1 -1
If a JavaMail messaage comes from some extant source (e.g.
eml file or over the wire), then

the transfer encoded version is available through
     getDataHandler().getRawInputStream()
the unencoded version is available through
     getContent() and getInputStream(), back by SharedInputStream

If a JavaMail messages is just being created, then

the transfer encoded version is not available at all
the unencoded version is available through
     getContent() and getInputStream() back by FileInputStream
     or ByteArrayInputStream or an underlying String.

Our method must always return the transfer encoded version.
----------------------------
revision 1.15
date: 2013/11/07 19:28:33;  author: barbee;  state: Exp;  lines: +1 -1
Fix filiaze from single only message to multipart message.
----------------------------
revision 1.14
date: 2013/10/25 13:47:15;  author: barbee;  state: Exp;  lines: +1 -1
Bug fix.
----------------------------
revision 1.13
date: 2013/10/15 20:02:03;  author: barbee;  state: Exp;  lines: +1 -1
Added default headers when Pantomime creates a brand new message.
Updated version.
----------------------------
revision 1.12
date: 2013/10/11 16:10:36;  author: barbee;  state: Exp;  lines: +1 -1
Update version.
----------------------------
revision 1.11
date: 2013/10/11 15:44:59;  author: barbee;  state: Exp;  lines: +6 -1
Switch to slf4j for logging.
----------------------------
revision 1.10
date: 2013/10/10 15:57:56;  author: barbee;  state: Exp;  lines: +1 -1
Updated version.
----------------------------
revision 1.9
date: 2013/10/02 14:43:06;  author: barbee;  state: Exp;  lines: +1 -1
Closed streams.

Updated version.

Added method to read body as string.
----------------------------
revision 1.8
date: 2013/09/27 18:14:07;  author: barbee;  state: Exp;  lines: +5 -5
Switched multipart and single part specific methods to inner classes
for better code access restrictions.

Added more javadocs.

Implemented dumpTree().
----------------------------
revision 1.7
date: 2013/09/19 19:17:22;  author: barbee;  state: Exp;  lines: +1 -1
Added Javadocs.

Moved Content classes into its own package.

Adjusted class and method modifiers to hide stuff.

Make naming of getSize() methods consistent.
----------------------------
revision 1.6
date: 2013/08/30 19:28:52;  author: barbee;  state: Exp;  lines: +1 -1
New version.
----------------------------
revision 1.5
date: 2013/08/29 20:49:10;  author: barbee;  state: Exp;  lines: +1 -1
new version
----------------------------
revision 1.4
date: 2013/08/29 20:48:36;  author: barbee;  state: Exp;  lines: +10 -0
Test blob message source.
----------------------------
revision 1.3
date: 2013/08/27 16:39:17;  author: barbee;  state: Exp;  lines: +5 -1
Added licensing.
Version is now 0.25.
----------------------------
revision 1.2
date: 2013/08/23 21:16:14;  author: barbee;  state: Exp;  lines: +1 -1
Done with 2007.
----------------------------
revision 1.1
date: 2013/08/21 19:25:23;  author: barbee;  state: Exp;
branches:  1.1.1;
Initial revision
----------------------------
revision 1.1.1.1
date: 2013/08/21 19:25:23;  author: barbee;  state: Exp;  lines: +0 -0
New project.
=============================================================================

RCS file: /home/barbee/CVS/pantomime/javadoc.sh,v
Working file: javadoc.sh
head: 1.4
branch:
locks: strict
access list:
symbolic names:
keyword substitution: kv
total revisions: 4;	selected revisions: 4
description:
----------------------------
revision 1.4
date: 2013/10/11 18:41:18;  author: barbee;  state: Exp;  lines: +2 -2
Switched logging.
Updated version number.
Added static free method.
----------------------------
revision 1.3
date: 2013/10/08 17:29:18;  author: barbee;  state: Exp;  lines: +3 -3
Fix.
----------------------------
revision 1.2
date: 2013/09/27 18:14:07;  author: barbee;  state: Exp;  lines: +23 -4
Switched multipart and single part specific methods to inner classes
for better code access restrictions.

Added more javadocs.

Implemented dumpTree().

src/main/java/org/blackmist/pantomime/Line.java
/**
 * Copyright (c) 2013-2015 <JH Barbee>
 * 
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 * 
 * Initial Developer: JH Barbee
 *
 * For support, please see https://bitbucket.org/barbee/pantomime
 * 
 * $Id: Line.java,v 1.2 2013/08/27 16:39:17 barbee Exp $
**/

package org.blackmist.pantomime;

class Line {
    long position;
    String text;
    String ending;

    public long getEndOfLinePosition() {
        long number = position;
        if ( text != null ) {
            number += text.length();
        }
        if ( ending != null ) {
            number += ending.length();
        }
        return number;
    }

    public String toString() {

        StringBuilder builder = new StringBuilder();

        builder.append("{ Position: ").append(position)
            .append(" Text: ");
        
        if ( text == null ) {
            builder.append("Null ");
        } else {
            int len = text.length();
            if ( len > 10 ) {
                len = 10;
            }
            builder.append("\"");
            builder.append(text.substring(0, len));
            builder.append("\" ");
            builder.append(text.length());
            builder.append(" chars ");
        }

        builder.append(" Ending: ");

        if ( ending == null ) {
            builder.append("Null");
        } else if ( ending.length() == 0 ) {
            builder.append("None");
        } else {
            for ( int index = 0; index < ending.length(); index++ ) {
                int num = ending.charAt(index);
                builder.append(num);
                builder.append(" ");
            }
        }

        builder.append(" }");

        return builder.toString();
    }
}

